geomstats/geometry/dirichlet_distributions.py

```diff
@@ -0,0 +1,322 @@
+
+
+class DirichletMetric(RiemannianMetric):
+    """Class for the Fisher information metric on Dirichlet distributions."""
+
+    def __init__(self, dim):
+        super(DirichletMetric, self).__init__(dim=dim)
+
+    def metric_matrix(self, base_point=None):
+        """Compute the inner-product matrix.
+
+        Compute the inner-product matrix of the Fisher information metric
+        at the tangent space at base point.
+
+        Parameters
+        ----------
+        base_point : array-like, shape=[..., dim]
+            Base point.
+
+        Returns
+        -------
+        mat : array-like, shape=[..., dim, dim]
+            Inner-product matrix.
+        """
+        if base_point is None:
+            raise ValueError('A base point must be given to compute the '
+                             'metric matrix')
+        base_point = gs.to_ndarray(base_point, to_ndim=2)
+        n_points = base_point.shape[0]
+
+        mat_ones = gs.ones((n_points, self.dim, self.dim))
+        poly_sum = gs.polygamma(1, gs.sum(base_point, -1))
+        mat_diag = gs.zeros((n_points, self.dim, self.dim))
+        diag = gs.einsum('ijj->ij', mat_diag)
+        diag[:] = gs.polygamma(1, base_point)
+
+        mat = mat_diag - gs.einsum('i,ijk->ijk', poly_sum, mat_ones)
+        return gs.squeeze(mat)
+
+    def christoffels(self, base_point):
+        """Compute the Christoffel symbols.
+
+        Compute the Christoffel symbols of the Fisher information metric.
+
+        Parameters
+        ----------
+        base_point : array-like, shape=[..., dim]
+            Base point.
+
+        Returns
+        -------
+        christoffels : array-like, shape=[..., dim, dim, dim]
+            Christoffel symbols.
+        """
+        base_point = gs.to_ndarray(base_point, to_ndim=2)
+        n_points = base_point.shape[0]
+
+        def coefficients(ind_k):
+            param_k = base_point[..., ind_k]
+            param_sum = gs.sum(base_point, -1)
+            C1 = 1 / gs.polygamma(1, param_k) / (
+                1 / gs.polygamma(1, param_sum)
+                - gs.sum(1 / gs.polygamma(1, base_point), -1))
+            C2 = - C1 * gs.polygamma(2, param_sum) / gs.polygamma(1, param_sum)
+
+            mat_ones = gs.ones((n_points, self.dim, self.dim))
+            mat_diag = gs.zeros((n_points, self.dim, self.dim))
+            diag = gs.einsum('ijj->ij', mat_diag)
+            diag[:] = - gs.polygamma(2, base_point) \
+                / gs.polygamma(1, base_point)
+
+            mat_k = gs.einsum('i,ijk->ijk', C2, mat_ones)\
+                - gs.einsum('i,ijk->ijk', C1, mat_diag)
+            mat_k[:, ind_k, ind_k] += gs.polygamma(2, param_k)\
+                / gs.polygamma(1, param_k)
+            return 1 / 2 * mat_k
+
+        christoffels = gs.zeros((n_points, self.dim, self.dim, self.dim))
+        for ind_k in range(self.dim):
+            christoffels[:, ind_k, :, :] = coefficients(ind_k)
```

Comment:
In case this could work for tensorflow in the future, it would be nice to avoid assignment already: for example construct `christoffels` as a list by appending the new entry in the loop, then stack along the second dimension.